src/glewlwyd.c

```diff
@@ -181,15 +185,24 @@ int main (int argc, char ** argv) {
       pthread_cond_init(&global_handler_close_cond, NULL)) {
     y_log_message(Y_LOG_LEVEL_ERROR, "init - Error initializing global_handler_close_lock or global_handler_close_cond");
   }
-  // Catch end signals to make a clean exit
-  if (signal (SIGQUIT, exit_handler) == SIG_ERR ||
-      signal (SIGINT, exit_handler) == SIG_ERR ||
-      signal (SIGTERM, exit_handler) == SIG_ERR ||
-      signal (SIGHUP, exit_handler) == SIG_ERR ||
-      signal (SIGBUS, exit_handler) == SIG_ERR ||
-      signal (SIGSEGV, exit_handler) == SIG_ERR ||
-      signal (SIGILL, exit_handler) == SIG_ERR) {
-    fprintf(stderr, "init - Error initializing end signal\n");
+
+  // Process end signals on dedicated thread
+  sigemptyset(&close_signals);
```

Comment:
I think we should test the result of `sigemptyset` and `sigaddset` to be 0, just to be sure

Reply:
Definitely!